pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -92,27 +94,26 @@ public static List<ExpressionContext> extractExpressions(QueryContext queryConte
     }
 
     List<ExpressionContext> selectExpressions = queryContext.getSelectExpressions();
-    if (selectExpressions.size() == 1 && selectExpressions.get(0).equals(IDENTIFIER_STAR)) {
-      // For 'SELECT *', sort all columns (ignore columns that start with '$') so that the order is deterministic
-      Set<String> allColumns = indexSegment.getColumnNames();
-      List<String> selectColumns = new ArrayList<>(allColumns.size());
-      for (String column : allColumns) {
-        if (column.charAt(0) != '$') {
-          selectColumns.add(column);
+    for (ExpressionContext selectExpression : selectExpressions) {
```

Comment:
This is unrelated. Let's make a separate PR for this.
We won't hit this when the schema is present because broker should already rewrite `*` to actual columns